starcheck/src/lib/Ska/Starcheck/Obsid.pm

```diff
@@ -2656,14 +2655,195 @@ sub set_ccd_temps{
+    my $cat_cmd = find_command($self, "MP_STARCAT");
+    if ((not $targ_cmd) or (not $cat_cmd) or ($self->{obsid} =~ /NONE(\d+)/)){
+        return \%proseco_args;
+    }
+    my $si = 'ACIS-S';
+    my $offset = 0;
+    if ($self->{obsid} < 38000){
+        $si = $self->{SI};
+        $offset = $self->{SIM_OFFSET_Z};
+    }
+    my $man_ang = 90;
+    if (defined $targ_cmd->{angle}){
+        $man_ang = $targ_cmd->{angle};
+    }
+
+    my @acq_ids;
+    my @acq_indexes;
+    my @gui_ids;
+    my @fid_ids;
+    my @halfwidths;
+  IDX:
+    foreach my $i (1..16) {
+        (my $sid  = $cat_cmd->{"GS_ID$i"}) =~ s/[\s\*]//g;
+        # If there is no star there is nothing for proseco probs to do so skip it.
+        # But warn if it was a thing that should have had an id (BOT/ACQ/GUI).
+        if ($sid eq '---'){
+            if ($cat_cmd->{"TYPE$i"} =~ /BOT|ACQ|GUI/){
+                push @{$self->{warn}}, sprintf(
+                    ">> WARNING: [%2d] Could not calculate acq prob for star with no id.", $i);
+            }
+            next IDX;
+        }
+        $sid = int($sid);
+        ## Strip off asterisk if present (there's probably a cleaner Perl way to do this).
+        #if ($sid =~ /\*(\d+)/){
+        #    $sid = $1;
+        #}
+        if ($cat_cmd->{"TYPE$i"} =~ /BOT|ACQ/){
+            push @acq_ids, $sid;;
+            my $hw = $cat_cmd->{"HALFW$i"};
+            if (($hw > 180) or ($hw < 60)){
+                push @{$self->{yellow_warn}}, sprintf(
+                    ">> WARNING: [%2d] Halfwidth %d outside range 60 to 180. Clipped for probs.",
+                    $i, $hw);
+                # Clip hw if outside the range 60 to 180 for probabilities
+                $hw =   $hw < 60  ? 60
+                      : $hw > 180 ? 180
+                      : $hw ;
+            }
+            push @halfwidths, $hw;
+            push @acq_indexes, $i;
+        }
+        if ($cat_cmd->{"TYPE$i"} =~ /BOT|GUI/){
+            push @gui_ids, $sid;
+        }
+        if ($cat_cmd->{"TYPE$i"} =~ /FID/){
+            push @fid_ids, $sid;
+        }
+    }
+
+    %proseco_args = (
+        obsid => $self->{obsid},
+        date => $targ_cmd->{stop_date},
+        q1 => 0 + $targ_cmd->{q1}, q2 => 0 + $targ_cmd->{q2}, q3 => 0 + $targ_cmd->{q3}, q4 =>0 + $targ_cmd->{q4},
+        man_angle => $man_ang,
+        detector => $si, offset=> 0 + $offset,
+        dither_acq_y => $self->{dither_acq}->{ampl_y},
+        dither_acq_z => $self->{dither_acq}->{ampl_p},
+        dither_guide_y => $self->{dither_guide}->{ampl_y},
+        dither_guide_z => $self->{dither_guide}->{ampl_p},
+        t_ccd_acq => $self->{ccd_temp_acq},
+        t_ccd_guide => $self->{ccd_temp},
+        acqs => \@acq_ids,
+        halfwidths => \@halfwidths,
+        fids => \@fid_ids,
+        guides => \@gui_ids,
+        acq_indexes => \@acq_indexes);
+
+    return \%proseco_args
+
+}
+
+my $CUM_PROB_LIMIT = 0.008;
```

Comment:
Did you want just P2=2.1 or P2=2.0 for starcheck 13.0 or a new value?

Reply:
P2 = 2.0 (0.01).